zipline/finance/trading.py

```diff
@@ -114,11 +124,11 @@ def __init__(
         if isinstance(asset_db_path, string_types):
             asset_db_path = 'sqlite:///%s' % asset_db_path
             self.engine = engine = create_engine(asset_db_path)
-            AssetDBWriterFromDictionary().init_db(engine)
         else:
             self.engine = engine = asset_db_path
 
         if engine is not None:
+            AssetDBWriter(engine).init_db()
```

Comment:
Does the difference in location mean that it's more robust to a db already being initialized?